src/Types/Input/EnumInputType.hack

```diff
@@ -0,0 +1,33 @@
+namespace Slack\GraphQL\Types;
+
+use namespace Graphpinator\Parser\Value;
+use namespace Slack\GraphQL;
+
+
+abstract class EnumInputType extends NamedInputType {
+
+    <<__Enforceable>>
+    abstract const type TCoerced as arraykey;
+    abstract const \HH\enumname<this::TCoerced> HACK_ENUM;
+
+    <<__Override>>
+    public function coerceValue(mixed $value): this::TCoerced {
+        if (!$value is this::TCoerced) {
```

Comment:
Do `is` and `as` replace `coerce` and `assert` from enums? I thought you'd need to use those but it looks like this works too.

Reply:
I think we should be expecting the enum const name here, not the internal value (which is what this check would accept). Probably something like

```Hack
$enum = static::HACK_ENUM;
GraphQL\assert(
  C\contains_key($enum::getValues(), $value),
  'Expected ...'
);
return $enum::getValues()[$value];
```

Reply:
Good point about `contains_key`, will update.